### PR TITLE
DPC-917 Add since parameter to patient/everything endpoint

### PR DESCRIPTION
**Pull request by PatKelsh**
### Fixes [DPC-917](https://jira.cms.gov/browse/DPC-917)

The `patient/everything` endpoint does not accept the `_since` parameter and users are not able to limit requests to only returning the most recent data

### Proposed Changes

* Added the `_since` parameter to the patient/everything endpoint
* Validated the `since` parameter and passed it on to the data service
* Updated tests to exercise the endpoint with the `_since` parameter

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation

Updated tests pass with no issues

### CODE REVIEW

File: dpc-queue/src/main/java/gov/cms/dpc/queue/service/DataService.java [hunk 1]
```diff
@@ -46,26 +46,28 @@ public DataService(IJobQueue queue, FhirContext fhirContext, @ExportPath String
...
      * @return Resource
      */
     public Resource retrieveData(UUID organizationID,
                                  UUID providerID,
                                  List<String> patientMBIs,
+                                 OffsetDateTime since,
                                  OffsetDateTime transactionTime,
                                  String requestingIP, ResourceType... resourceTypes) {
-        UUID jobID = this.queue.createJob(organizationID, providerID.toString(), patientMBIs, List.of(resourceTypes), null, transactionTime, requestingIP, false);
-        LOGGER.info("Patient everything export job created with job_id={}",jobID.toString());
```
Comment (jonfulk) [inline dpc-queue/src/main/java/gov/cms/dpc/queue/service/DataService.java:67]: Did you remove this logging statement on purpose?

File: dpc-api/src/test/java/gov/cms/dpc/api/resources/v1/PatientResourceTest.java [hunk 2]
```diff
@@ -399,6 +511,10 @@ public void testPatientPathAuthorization() throws GeneralSecurityException, IOEx
 
     }
 
+    public void testPatientEverything() {
+
+    }
+
```
Comment (jonfulk) [inline dpc-api/src/test/java/gov/cms/dpc/api/resources/v1/PatientResourceTest.java:517]: I think you probably meant to delete this

File: dpc-api/src/test/java/gov/cms/dpc/api/resources/v1/PatientResourceUnitTest.java [hunk 3]
```diff
@@ -14,6 +14,9 @@
 import org.mockito.Mock;
 import org.mockito.Mockito;
 import org.mockito.MockitoAnnotations;
+import javax.servlet.http.HttpServletRequest;
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.Provenance;
```
Comment (jonfulk) [inline dpc-api/src/test/java/gov/cms/dpc/api/resources/v1/PatientResourceUnitTest.java:19]: unused imports?

**Review by jonfulk — COMMENTED** [review]
I know this is still in draft, but I left a few comments for you anyway :)

File: dpc-api/src/main/java/gov/cms/dpc/api/resources/AbstractPatientResource.java [hunk 4]
```diff
@@ -39,7 +39,11 @@ protected AbstractPatientResource() {
 
     @GET
     @Path("/{patientID}/$everything")
-    public abstract Resource everything(OrganizationPrincipal organization, @Valid @Profiled(profile = AttestationProfile.PROFILE_URI) Provenance attestation, UUID patientId, HttpServletRequest request);
+    public abstract Resource everything(OrganizationPrincipal organization,
```
[the remark this answers is not in the document]
Comment (PatKelsh) [reply dpc-api/src/main/java/gov/cms/dpc/api/resources/AbstractPatientResource.java:43]: All of these arguments are needed by the function. I think the effort to reduce the number of arguments would make the code harder to read and not provide any benefit

**Review by jonfulk — APPROVED** [review]

File: dpc-queue/src/main/java/gov/cms/dpc/queue/service/DataService.java (hunk 1 above)
Comment (dhgreene) [reply dpc-queue/src/main/java/gov/cms/dpc/queue/service/DataService.java:67]: Do we want to log the `_since` param [like we do elsewhere](https://github.com/CMSgov/dpc-app/blob/8099f0788646b571cb5a397f177cf698620996d5/dpc-api/src/main/java/gov/cms/dpc/api/resources/v1/GroupResource.java#L286)?

**Review by dhgreene — DISMISSED** [review]
🏁

File: dpc-api/src/test/java/gov/cms/dpc/api/resources/v1/PatientResourceUnitTest.java [hunk 5]
```diff
@@ -28,6 +29,9 @@
     @Mock
     FhirValidator fhirValidator;
 
+    @Mock
+    HttpServletRequest request;
+
```
Comment (SalGutierrez) [inline dpc-api/src/test/java/gov/cms/dpc/api/resources/v1/PatientResourceUnitTest.java:34]: Is this still needed or just snuck in?

**Review by SalGutierrez — APPROVED** [review]
Small comment left, but LGTM! 🚀

File: dpc-api/src/test/java/gov/cms/dpc/api/resources/v1/PatientResourceUnitTest.java [hunk 6]
```diff
@@ -14,6 +14,7 @@
 import org.mockito.Mock;
 import org.mockito.Mockito;
 import org.mockito.MockitoAnnotations;
+import javax.servlet.http.HttpServletRequest;
```
Comment (jonfulk) [inline dpc-api/src/test/java/gov/cms/dpc/api/resources/v1/PatientResourceUnitTest.java:17]: unused import

**Review by jonfulk — APPROVED** [review]
Just one unused import. if you took that out, it's G2G!

**Pull request merged**